model/pipeline/src/main/resources/org/apache/beam/model/pipeline/pipeline.json

```diff
@@ -0,0 +1,320 @@
+        // can be used for replacement-based implementation and also to parse the node
+        // to a concrete language-specific class.
+        //
+        // A stable, cross-language identifier identifying what the transform computes.
+        // This is to be used for replacement-based implementation, whether it is a primitive
+        // or composite. Examples include "GroupByKey", "Sum", "ParDo". We should likely
+        // adopt a standard namespacing convention, such as URIs.
+        "urn":  { "type": [ "null", "string" ] },
+
+        // All of the inputs and outputs for the transform, whether it is composite or primitive.
+        // The inputs and outputs of parent and child transforms nodes must be coherent:
+        //
+        // - The unexpanded inputs of every child transforms must be contained
+        //   within some level of the recursive expansion of the inputs to the parent.
+        // - Every level of the expanded output of a composite must be contained within
+        //   some level of the expansion of some child.
+        //
+        // These collections of values are unordered. References to values are managed in
+        // a transform-specific way. For example, a DoFn will call processContext.output(tag, value)
+        // and the tag should contain sufficient information to identify the target PCollection.
+        "inputs": { "type": "array", "items": { "type": "string" } },
+        "outputs": { "type": "array", "items": { "type": "string" } },
+
+        // If the node is contained within a composite, a pointer to the parent.
+        "parent": { "type": [ "null", "string" ] },
+
+        // If the node is a runner-specific replacement for a user's runner-agnostic node,
+        // it contains a pointer back to it so the user's graph can be reconstructed.
+        "replacementFor": { "type": [ "null", "string" ] },
+
+        // If the node is a defunct user node that has been replaced, it contains a bit
+        // to distinguish it from an active node in the graph.
+        "replaced": { "type": "boolean" },
+
+        // Static display data
+        "display_data": { "type": "object", "additionalProperties": { "type": "string" } },
+
+        // The payload must correspond to the kind string and must be null for transforms that
+        // are not predefined as part of the Beam model.
+        "payload": {
+          "type": [ "null", "object" ],
+          "oneOf": [
+            { "$ref": "#/definitions/ParDoPayload" },
+            { "$ref": "#/definitions/ReadPayload" },
+            { "$ref": "#/definitions/WindowIntoPayload" },
+            { "$ref": "#/definitions/CombinePayload" }
+          ]
+        }
+      }
+    },
+
+    // A value node represents a PValue. There are only two primitive types:
+    // A PCollection (many values per window) or a PCollectionView (one value per window).
+    "Value": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
```

Comment:
Would it need to have a type identifying whether this is a `PCollection` or a `PCollectionView`?